internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -2009,6 +2010,27 @@ func clusterCompleted(clusterInstall *hiveext.AgentClusterInstall, clusterDeploy
 	})
 }
 
+func lastInstallationPreparationFailed(clusterInstall *hiveext.AgentClusterInstall, installation_preparation_completion_status string, installation_preparation_completion_status_reason string) {
+	var condStatus corev1.ConditionStatus = corev1.ConditionFalse
+	var reason string = hiveext.ClusterLastInstallationPreparationFailedOKReason
+	var msg string = ""
+
+	switch installation_preparation_completion_status {
```

Comment:
Please let the switch-case handle all possible cases.  It is not clear from the code.